### PR TITLE
ENH/POC: infer resolution in array_strptime

### PR DESCRIPTION
**Pull request by jbrockmendel**
array_strptime analogue to #55741, en route to #55564.

### CODE REVIEW

File: pandas/_libs/tslibs/strptime.pyx [hunk 1]
```diff
@@ -232,9 +237,21 @@ cdef _get_format_regex(str fmt):
...
+        self.creso = creso
+        self.creso_changed = False
+
+    cdef bint update_creso(self, NPY_DATETIMEUNIT item_reso) noexcept:
+        # Return a bool indicating whether we bumped to a higher resolution
+        if self.creso == NPY_DATETIMEUNIT.NPY_FR_GENERIC:
+            self.creso = item_reso
+        elif item_reso > self.creso:
+            self.creso = item_reso
+            self.creso_changed = True
```
Comment (mroeschke) [inline pandas/_libs/tslibs/strptime.pyx:252]: So we only want to track if this changes once from the default? e.g.

```
>>> s = DatetimeParseState("s")
>>> s.update_creso("ms")
>>> s.update_creso("ms")
>>> s.creso_changed  # should still return True?
```

Comment (jbrockmendel) [reply pandas/_libs/tslibs/strptime.pyx:252]: correct.  this is a has-it-ever-changed-since-init

Comment (mroeschke) [reply pandas/_libs/tslibs/strptime.pyx:252]: Gotcha. Could we rename this like `creso_changed_from_default`?

Comment (jbrockmendel) [reply pandas/_libs/tslibs/strptime.pyx:252]: sure, will update

Comment (jbrockmendel) [reply pandas/_libs/tslibs/strptime.pyx]: updated

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**